src/Decoder/SmileDecoder.php

```diff
@@ -18,11 +18,17 @@ public function decode(string $smileData): array|\stdClass
         $this->context = new SmileDecoderContext(unpack('C*', $smileData));
         $this->decodeHead();
 
-        return match ($this->getNextByte()) {
-            Bytes::LITERAL_ARRAY_START => $this->decodeArray(),
-            Bytes::LITERAL_OBJECT_START => $this->decodeObject(),
-            default => throw new UnexpectedValueException('The smile file seems invalid since it doesn\'t start with an array or an object.'),
-        };
+        try {
+            $result = match ($this->getNextByte()) {
```

Comment:
You can return here directly

Reply:
This will be done in the next PR, which will clean the project